# Large-scale integrative analysis of soybean transcriptome using an unsupervised autoencoder model
===
# Introduction
Plant tissues are distinguished by their gene expression patterns, which can help identify tissue-specific highly expressed genes and their differential functional modules. For this purpose, large-scale soybean transcriptome samples were collected and processed starting from the raw sequencing reads in a uniform analysis pipeline. To address the gene expression heterogeneity in different tissues, we utilized an adversarial deconfounding autoencoder (AD-AE) model to map gene expression into a latent space, and adapted a standard unsupervised autoencoder (AE) model to help effectively extract meaningful biological signals from the noisy data. As a result, 1,743, 914, 2,107 and 1,451 genes were found highly expressed specifically in leaf, root, seed and nodule tissues, respectively. To obtain key transcription factors (TFs), hub genes and their functional modules in each tissue, tissue-specific gene regulatory networks and differential correlation networks were constructed by using corrected and compressed gene expression data. We validated our results from the literature and gene enrichment analysis, which confirmed many identified tissue-specific genes. Our study represents the largest gene expression analysis in soybean tissues to date. It provides valuable targets for tissue-specific research and helps uncover broader biological patterns. 
# Adversarial Deconfounding Autoencoder 
## Introduced by Paul G. Allen School of Computer Science & Engineering, University of Washington, Seattle
The ADAE (Adversarial Deconfounding AutoEncoder) approach to deconfound the gene expression latent spaces. The model consists of two neural networks: (i) an autoencoder to generate an embedding that can successfully reconstruct the original measurements and (ii) an adversary that is trained to predict the confounder from the embedding. The idea is to jointly train the networks to generate embeddings that can encode as much information as possible while not being able to predict the confounders. By applying AD-AE to two distinct gene expression datasets, we show that our model can (1) generate embeddings that do not encode confounder information, (2) conserve the biological signals present in the original space, and (3) generalize successfully across different confounder domains. We demonstrate that AD-AE can outperform standard autoencoder as well as other deconfounding approaches.
![](https://github.com/LingtaoSu/SoyMeta/blob/master/Pipeline.jpg)

